src/GafferScene/Instancer.cpp

```diff
@@ -2356,3 +2569,240 @@ void Instancer::PrototypeScope::setPrototype( const EngineData *engine, const Sc
 		set( ScenePlug::scenePathContextName, prototypeRoot );
 	}
 }
+
+namespace
+{
+
+void addPrototypeToVectors(
+	const ScenePlug *prototypesPlug, const ScenePlug::ScenePath *prototypeRoot, Context::EditableScope &scope,
+	const std::vector<float> &sampleTimes, const IECore::MurmurHash &hash, int index,
+	std::vector<ObjectPtr> &objects, std::vector<M44f> &transforms, std::vector<ObjectPtr> &attributes
+)
+{
+	float onFrameTime = scope.context()->getFrame();
+
+	scope.set( ScenePlug::scenePathContextName, prototypeRoot );
+
+	// \todo - are there situations where this will be slow, and the renderer doesn't use it?
+	Box3f bound = prototypesPlug->boundPlug()->getValue();
+
+	// const_cast should be safe since the elements of the attributes vector are never modified
+	attributes.push_back( const_cast< CompoundObject*>( prototypesPlug->attributesPlug()->getValue().get() ) );
+
+	for( unsigned int i = 0; i < sampleTimes.size(); i++ )
+	{
+		scope.setFrame( sampleTimes[i] );
+		transforms.push_back( prototypesPlug->transformPlug()->getValue() );
+	}
+
+	scope.remove( ScenePlug::scenePathContextName );
+
+	IECore::MurmurHash h = hash;
+	h.append( index );
+
+	// We find the capsules using the engine at shutter open, but the time used to construct the capsules
+	// must be the on-frame time, since the capsules will add their own shutter
+	scope.setFrame( onFrameTime );
+
+
+	objects.push_back( new Capsule(
+		prototypesPlug,
+		*prototypeRoot,
+		*scope.context(),
+		h,
+		bound
+	) );
+
+
+}
+
+struct MurmurHashHasher
+{
+	std::size_t operator()(const IECore::MurmurHash& h) const
+	{
+		return h.h1();
+	}
+};
```

Comment:
Could you put it on your list to open a PR to specialise `std::hash<MurmurHash>` in Cortex please, and add a todo here to use it when it trickles through? I've wanted this a bunch of times. Note that we already have specialisations for `boost::hash_value` and `tbb_hasher` in there, and they do something different to this - might be worth figuring out which is best.